src/services/release.ts

```diff
@@ -186,5 +188,47 @@ export async function publishActiveRelease(): Promise<Release> {
   release = await releaseService.publishActiveRelease();
   logger.debug(`[Release.Publish] Release marked as published.`);
 
+  const message = buildKafkaMessage(release, filesAdded, filesRemoved);
+  sendPublicReleaseMessage(message.toString());
+
   return release;
 }
+
+const buildKafkaMessage = (
+  release: Release,
+  filesAdded: File[],
+  filesRemoved: File[],
+): PublicReleaseMessage => {
+  const filesAddedByProgramId = _.groupBy(filesAdded, file => file.programId);
+  const filesRemovedByProgramId = _.groupBy(filesRemoved, file => file.programId);
+
+  // merge removed and added files by programId into filesAddedByProgramId:
+  Object.entries(filesRemovedByProgramId).forEach(([programId, files]) => {
+    const existing = filesAddedByProgramId[programId] ? filesAddedByProgramId[programId] : [];
+    filesAddedByProgramId[programId] = [...existing, ...files];
+  });
+
+  // get unique donor ids from filesAdded and filesRemoved:
+  const programsUpdated: Program[] = [];
+
+  Object.entries(filesAddedByProgramId).forEach(([programId, files]) => {
+    const donorIds = new Set();
```

Comment:
`new Set<string>()` to make typescript work, then you dont need the `as string[]` below.